web/src/components/zonemap.jsx

```diff
@@ -107,22 +106,62 @@ const ZoneMap = ({
       hover: { 'fill-color': 'white', 'fill-opacity': 0.3 },
       ocean: { 'background-color': theme.oceanColor },
       zonesBorder: { 'line-color': theme.strokeColor, 'line-width': theme.strokeWidth },
-      zonesClickable: { 'fill-color': ['case', ['has', 'color'], ['get', 'color'], theme.clickableFill] },
+      zonesClickable: {
+        'fill-color': [
+          'coalesce', // // https://docs.mapbox.com/mapbox-gl-js/style-spec/expressions/#coalesce
+          ['feature-state', 'color'],
+          ['get', 'color'],
+          theme.clickableFill,
+        ],
```

Comment:
I wasn't aware that mapbox was so advanced, but the coalesce feature is perfect for this use case!
Here it's basically saying:
- use feature-state if available
- else use color property
- fallback to gray color from theme

Reply:
Makes sense!